metadata-ingestion/src/datahub/ingestion/source/sql/hive/hive_source.py

```diff
@@ -0,0 +1,367 @@
+        rows = self._get_table_columns(connection, table_name, schema)
+        # Strip whitespace
+        rows = [[col.strip() if col else None for col in row] for row in rows]
+        # Filter out empty rows and comment
+        rows = [row for row in rows if row[0] and row[0] != "# col_name"]
+        result = []
+        for col_name, col_type, _comment in rows:
+            # Handle both oss hive and Databricks' hive partition header, respectively
+            if col_name in ("# Partition Information", "# Partitioning"):
+                break
+            # Take out the more detailed type information
+            # e.g. 'map<int,int>' -> 'map'
+            #      'decimal(10,1)' -> decimal
+            orig_col_type = col_type  # keep a copy
+            col_type = re.search(r"^\w+", col_type).group(0)  # type: ignore
+            try:
+                coltype = _type_map[col_type]
+            except KeyError:
+                util.warn(
+                    "Did not recognize type '{}' of column '{}'".format(
+                        col_type, col_name
+                    )
+                )
+                coltype = types.NullType  # type: ignore
+            result.append(
+                {
+                    "name": col_name,
+                    "type": coltype,
+                    "nullable": True,
+                    "default": None,
+                    "full_type": orig_col_type,  # pass it through
+                    "comment": _comment,
+                }
+            )
+        return result
+
+    DatabricksPyhiveDialect.get_columns = dbapi_get_columns_patched
+except ModuleNotFoundError:
+    pass
+except Exception as exp:
+    logger.warning(f"Failed to patch method due to {exp}")
+
+
+@reflection.cache  # type: ignore
+def get_view_names_patched(self, connection, schema=None, **kw):
+    query = "SHOW VIEWS"
+    if schema:
+        query += " IN " + self.identifier_preparer.quote_identifier(schema)
+    return [row[0] for row in connection.execute(query)]
+
+
+@reflection.cache  # type: ignore
+def get_view_definition_patched(self, connection, view_name, schema=None, **kw):
+    full_table = self.identifier_preparer.quote_identifier(view_name)
+    if schema:
+        full_table = "{}.{}".format(
+            self.identifier_preparer.quote_identifier(schema),
+            self.identifier_preparer.quote_identifier(view_name),
+        )
+    # Hive responds to the SHOW CREATE TABLE with the full view DDL,
+    # including the view definition. However, for multiline view definitions,
+    # it returns multiple rows (of one column each), each with a part of the definition.
+    # Any whitespace at the beginning/end of each view definition line is lost.
+    rows = connection.execute(f"SHOW CREATE TABLE {full_table}").fetchall()
+    parts = [row[0] for row in rows]
+    return "\n".join(parts)
+
+
+HiveDialect.get_view_names = get_view_names_patched
+HiveDialect.get_view_definition = get_view_definition_patched
```

Comment:
HiveDialect, this comes from our fork, right? https://github.com/acryldata/PyHive
May we just fix there?